WG_GO_CORE.md

```diff
@@ -0,0 +1,177 @@
+   * There are examples and documentation for ipfs users to create private-content dapps using IPFS (aka handling encryption on the client side)
+* `M (PX)`: IPFS is usable for building wordpress-level decentralized sites
+   * API doesn't suck (and is fast)
+   * We can build an interactive blogging engine with a built-in editor/publisher entirely in a webapp (backed by a go-ipfs instance). And it's *easy* to do this.
+
+### 🤝 IPFS Contributors and Developers
+
+Developers using IPFS can rely on go-ipfs as a platform on which to build their product, app, or ecosystem, and new contributors are well supported.
+
+* `M (P0)`: (all) Go-IPFS is approachable as a new contributor
+   * Go-IPFS internals are well documented
+   * Technical debt is paid off, all "indefinitely-in-progress" refactors are completed.
+   * Cleanup/rethink abstraction layers with 20-20 hindsight.
+   * Re-structure repos into reusable but not fragmented components.
+* `M (P0)`: (all) Go-IPFS is usable without gx
+   * Fix the versions (make them all sub-0)
+   * Add go.mod files to every package
+   * Merge #5435.
+* `M (P0)`: (all) Users can reliably transfer data between any two nodes.
+   * Reliable NAT traversal (AutoNAT, relay, TURN, etc).
+   * Connection manager doesn't kill useful connections.
+   * Scalable content routing
+   * Reliable DHT: unreliable nodes don't join the DHT.
+* `M (P0)`: (dapps) ipfs:// and ipns:// work in web browsers
+   * Base32 CIDs
+   * Base32 IPNS or IPNS uses CIDs
+* `M (P1)`: (all) Go-IPFS is approachable as a user/app developer
+   * User documentation.
+   * Well designed API interface
+   * Well designed API transport
+* `M (P1)`: (Dapps) Fast (< 3s), mutable name resolution (IPNS)
+   * Reliable DHT.
+   * QUIC (for fast connection establishment).
+   * Better protocol negotiation (multistream-2.0)
+   * Delegated Routing
+* `M (P1)`: (Dapps) IPFS Realtime Story is as good as using a centralized service (e.g. Socket.io, Pusher, etc)
+   * The API for doing secure (authenticated) broadcast updates exists 
+* `M (P2)`: Go-ipfs can handle large datasets (>1TiB, >1M nodes)
+   * Scalable content routing (providing)
+   * DagSync (or at least better bitswap)
+   * Reliable, performant datastore
+* `M (P2)`: (all, Dapps) IPFS is pluggable and extensible (9/10 users don't need custom features)
+   * The go-ipfs plugins system is expanded to support new datastores, exchanges, etc
+   * Exchanges support multiple protocols (that diagram from hack week)
+   * Multi-DHT (https://github.com/ipfs/notes/issues/291#issuecomment-414495124)
+   * UnixFS-V2 is implemented
+* `M (P2)`: (machine-learning/package managers) IPFS can handle (and transfer) large (>1M entries) sharded indexes (objects, directories)
+   * Bitswap Improvements (sessions, prediction, etc)
+   * UnixFS-V2 (better directory structure)
+* `M (P3)`: (Dapps) IPFS can locally share data without a shared network

```

Comment:
Creating an RFP in Q1 would allow us to find folks interested in working on this in Q2, allowing others to build on this capability in H2.
```suggestion
* `M (P3)`: (Dapps) IPFS can locally share data without a shared network
   * Q1 - Draft RFP for bluetooth (or like) transport and find owner
   * Q2 - Implement bluetooth (or like) transport
```